compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -1638,11 +1638,7 @@ public void visit(BLangUserDefinedType userDefinedType) {
             return;
         }
         BTypeSymbol tsymbol = resolvedType.tsymbol;
-        BSymbol pop = this.currDependentSymbolDeque.pop();
-        this.currDependentSymbolDeque.push(tsymbol);
-        recordGlobalVariableReferenceRelationship(pop);
-        this.currDependentSymbolDeque.pop();
-        this.currDependentSymbolDeque.push(pop);
+        recordGlobalVariableReferenceRelationship(tsymbol);
```

Comment:
With this change, we no longer log an error for

```ballerina
type Foo record {
    typedesc i = x;
};

final typedesc x = Foo;
```

That's not correct, right?

Reply:
I think it is not because of this change. Even without the change, error does not log

Reply:
Seems to be a regression master. I get the error on slbeta4 also. Created https://github.com/ballerina-platform/ballerina-lang/issues/34097.

I'm still not sure if this is the correct place for the change though. Isn't the issue we are trying to solve specifically applicable to service declarations? Looping in @rdhananjaya also.

Reply:
I think the issue is when dataflow analyzer sees a reference to a used-defined type, they way it records the dependency is wrong